code/game/gamemodes/dynamic/dynamic.dm

```diff
@@ -630,119 +557,112 @@ GLOBAL_VAR_INIT(dynamic_forced_storyteller, null)
 		// Check if a blocking ruleset has been executed.
 		else if(check_blocking(new_rule.blocking_rules, executed_rules))
 			return FALSE
-		// Check if the ruleset is highlander and if a highlander ruleset has been executed
-		else if(new_rule.flags & HIGHLANDER_RULESET)
-			if(threat_level > GLOB.dynamic_stacking_limit && GLOB.dynamic_no_stacking)
-				if(highlander_executed)
+		// Check if the ruleset is high impact and if a high impact ruleset has been executed
+		else if(new_rule.flags & HIGH_IMPACT_RULESET)
+			if(threat_level < GLOB.dynamic_stacking_limit && GLOB.dynamic_no_stacking)
+				if(high_impact_ruleset_executed)
 					return FALSE
 
-	update_playercounts()
-	if ((forced || (new_rule.acceptable(current_players[CURRENT_LIVING_PLAYERS].len, threat_level) && new_rule.cost <= threat)))
+	var/population = current_players[CURRENT_LIVING_PLAYERS].len
+	if((new_rule.acceptable(population, threat_level) && new_rule.cost <= mid_round_budget) || forced)
 		new_rule.trim_candidates()
 		if (new_rule.ready(forced))
-			log_threat("[new_rule.ruletype] - <b>[new_rule.name]</b> [new_rule.cost] threat", verbose = TRUE)
+			spend_midround_budget(new_rule.cost)
+			threat_log += "[worldtime2text()]: Forced rule [new_rule.name] spent [new_rule.cost]"
+			new_rule.pre_execute(population)
 			if (new_rule.execute()) // This should never fail since ready() returned 1
-				if(new_rule.flags & HIGHLANDER_RULESET)
-					highlander_executed = TRUE
+				if(new_rule.flags & HIGH_IMPACT_RULESET)
+					high_impact_ruleset_executed = TRUE
 				else if(new_rule.flags & ONLY_RULESET)
 					only_ruleset_executed = TRUE
 				log_game("DYNAMIC: Making a call to a specific ruleset...[new_rule.name]!")
-				SSblackbox.record_feedback("associative","dynamic_rulesets",1,new_rule.get_blackbox_info())
```

Comment:
same, if you removed get_blackbox_info don't bother

Reply:
Also gone